main.go

```diff
@@ -4,28 +4,47 @@ import (
 	"fmt"
 	"math/big"
 	"os"
+	"strconv"
 )
 
 var big0 = big.NewInt(0)
 var big1 = big.NewInt(1)
 
-func factorial(x *big.Int) *big.Int {
-	if x.Cmp(big0) == 0 {
-		return big1
+func factorial(x int) *big.Int {
+	vals := make(chan *big.Int, x+1)
+	res := make(chan *big.Int)
+	count := make(chan int)
+	for i := 1; i <= x; i++ {
+		j := big.NewInt(int64(i))
+		vals <- j
 	}
-	var res, xMinOne big.Int
-	return res.Mul(x, factorial(xMinOne.Sub(x, big1)))
+	for i := 0; i < x; i++ {
+		go func() {
```

Comment:
i'd consider changing a little bit the approach. In this case, you are starting x goroutines (one per process?). I'd consider just having a few that, in an "infinite" loop,  processes data as soon as they are in the channel (and waits until more data arrives)

Reply:
good idea :+1: